scripts/host-check

```diff
@@ -1104,6 +1104,22 @@ def check_iommu() -> CheckFuncReturn:
             f"Unable to check if IOMMU is enabled by listing {iommu_dev_filepath}.\n"
             + recommendation,
         )
+    return (CheckState.SUCCESS, "IOMMU enabled for vfio-pci.")
+
+
+def check_pci_devices() -> CheckFuncReturn:
```

Comment:
Per the PR description:

If IOMMU is enabled, checks that IOMMU PCI devices are available.
If IOMMU is not enabled, checks that any PCI devices are available.

I'm not convinced this is the desirable behaviour.  Instead should this be:

- If uio is present: check any PCI devices are available and list them all (customer can use uio and not worry about IOMMU; this is the same as `check_iommu`, which returns NEUTRAL if uio and vfio are present but IOMMU is disabled)
- If only vfio is present: check IOMMU PCI devices are available and list them, if not return WARNING.

Reply:
Yeah, I think that sounds right.

Reply:
Added a new require_iommu flag which is set to true if vfio is supported and uio is not, otherwise false. Then the check only requires IOMMU PCI devices if this flag is true.

Reply:
> except for the case that vfio is explicitly in non-IOMMU mode

This sounds reasonable - except there's now a bit of an inconsistency in that if we are using vfio only, and there are no `iommu_devices` but `enable_unsafe_noiommu_mode` is N or not present, then `check_iommu` is WARNING, and `check_pci_devices` is also WARNING ("No PCI network devices found with IOMMU enabled for vfio-pci").

Either this case should also list all PCI devices, or we should just list IOMMU PCI devices always for vfio only (I don't have a strong preference but lean slightly towards the latter).

The other option which maybe makes more sense from an end-user point of view would be to SKIP `check_pci_devices` if vfio only and `check_iommu` was WARN - but this is too much of a design change (SKIP is treated as a failure)

Reply:
(see `test_no_iommu_devices` :))

Reply:
Summary of offline discussion:
- If `/sys/module/vfio/parameters/enable_unsafe_noiommu_mode` is Y, then we assume the user knows what they are doing and its more useful for them to see all PCI devices.
- In contrast, there may be some IOMMU devices (so `check_iommu` passes), but no IOMMU netdevices.  In this case we shouldn't list the non-IOMMU netdevices.

Keep the behaviour as proposed - list IOMMU netdevices, unless the `/sys/module/vfio/parameters/enable_unsafe_noiommu_mode` is Y, in which case the user knows what they are doing (and they are already warned against doing this in `check_iommu`), so give them all netdevices.